tests/unit-tests/gpartition/test_hard_negative_post_partition.py

```diff
@@ -0,0 +1,114 @@
+"""
+    Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+"""
+import os
+import json
+import torch as th
+import numpy as np
+from typing import Dict
+
+import pytest
+
+from numpy.testing import assert_almost_equal
+from graphstorm.model.utils import load_dist_nid_map
+from dgl.data.utils import load_tensors, save_tensors
+from graphstorm.gpartition.post_hard_negative import (shuffle_hard_negative_nids,
+                                                      load_hard_negative_config)
+
+
+def test_load_hard_negative_config():
+    # For config with hard negative transformation
+    json_file_path = (f"/graphstorm/tests/unit-tests/gpartition/"
+                      f"config/gsprocessing_hard_negative_config.json")
+
+    res = load_hard_negative_config(json_file_path)
+
+    assert res[0] == {'dst_node_type': 'paper', 'edge_type':
+        'author:writing:paper', 'hard_neg_feat_name': 'hard_neg_feat'}
+
+    # For config without hard negative transformation
+    json_file_path = (f"/graphstorm/tests/unit-tests/gpartition/"
+                      f"config/gsprocessing_non_hard_negative_config.json")
+
+    res = load_hard_negative_config(json_file_path)
+
+    assert res == []
+
+
+def test_shuffle_hard_negative_nids(tmp_path):
```

Comment:
There's quite a bit of setup for this test is there any way to simplify, or at least move the setup in a fixture so we can focus on the test functionality?

Reply:
Let us make it a fixture.